libbpf-rs/src/program.rs

```diff
@@ -344,4 +354,23 @@ impl Program {
             Ok(Link::new(ptr))
         }
     }
+
+    /// Attach this program to [sk_lookup](https://lwn.net/Articles/819618/)
+    pub fn attach_sk_lookup(&mut self, target_fd: i32) -> Result<()> {
+        let mut opts = libbpf_sys::bpf_link_create_opts::default();
+        opts.sz = std::mem::size_of::<libbpf_sys::bpf_link_create_opts>() as u64;
+        let fd = unsafe {
```

Comment:
Do we need to return back the fd to the user so that the bpf link may be closed when the user is done with it?

Reply:
Yeah, It has to return Result<Link>

But I also see that libbpf provides bpf_program__attach_netns() API for this, so let's stay consistent?

Reply:
This is much better indeed. I just pushed an update to rely on `bpf_program__attach_netns` & return a `Result<Link>`. Thanks for the help!